src/app.py

```diff
@@ -1953,6 +1955,73 @@ def get_dataset_revision_number(id):
     # Response with the integer
     return jsonify(revision_number)
 
+@app.route('/datasets/<id>/revisions', methods=['GET'])
+def get_revisions_list(id):
+    # Token is not required, but if an invalid token provided,
+    # we need to tell the client with a 401 error
+    validate_token_if_auth_header_exists(request)
+
+    # Use the internal token to query the target entity
+    # since public entities don't require user token
+    token = get_internal_token()
+
+    # Query target entity against uuid-api and neo4j and return as a dict if exists
+    entity_dict = query_target_entity(id, token)
+    normalized_entity_type = entity_dict['entity_type']
+
+    # Only for Dataset
+    if normalized_entity_type != 'Dataset':
+        bad_request_error("The entity of given id is not a Dataset")
+
+    # Only published/public datasets don't require token
+    if entity_dict['status'].lower() != DATASET_STATUS_PUBLISHED:
+        # Token is required and the user must belong to HuBMAP-READ group
+        token = get_user_token(request, non_public_access_required=True)
+
+    # By now, either the entity is public accessible or
+    # the user token has the correct access level
+
+    complete_dict = schema_manager.get_complete_entity_result(token, entity_dict)
+    final_result = schema_manager.normalize_entity_result_for_response(complete_dict)
+    uuid = final_result['uuid']
+    previous_revisions = app_neo4j_queries.get_previous_revisions(neo4j_driver_instance, uuid)
+    complete_previous_list = schema_manager.get_complete_entities_list(token, previous_revisions)
+    next_revisions = app_neo4j_queries.get_next_revisions(neo4j_driver_instance, uuid)
+    complete_next_list = schema_manager.get_complete_entities_list(token, next_revisions)
+    outputjson = []
+    total_revisions = 1
+    for i in complete_previous_list:
+        revision = collections.OrderedDict()
```

Comment:
I don't see the need to use this `collections.OrderedDict()` because we don't care about the ordering of these three properties: `dataset`, `dataset_uuid`, and `revision_number`. And they should be consistent without enforcing ordering from the result anyway.